server/v2/cometbft/handlers/defaults.go

```diff
@@ -0,0 +1,148 @@
+package handlers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	consensusv1 "cosmossdk.io/api/cosmos/consensus/v1"
+	corecontext "cosmossdk.io/core/context"
+	"cosmossdk.io/core/transaction"
+	"cosmossdk.io/server/v2/core/appmanager"
+	"cosmossdk.io/server/v2/core/mempool"
+	"cosmossdk.io/server/v2/core/store"
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cosmos/gogoproto/proto"
+)
+
+type DefaultProposalHandler[T transaction.Tx] struct {
+	mempool    mempool.Mempool[T]
+	txSelector TxSelector[T]
+}
+
+func NewDefaultProposalHandler[T transaction.Tx](mp mempool.Mempool[T]) *DefaultProposalHandler[T] {
+	return &DefaultProposalHandler[T]{
+		mempool:    mp,
+		txSelector: NewDefaultTxSelector[T](),
+	}
+}
+
+func (h *DefaultProposalHandler[T]) PrepareHandler() appmanager.PrepareHandler[T] {
+	return func(ctx context.Context, app appmanager.AppManager[T], rm store.ReaderMap, txs []T, req proto.Message) ([]T, error) {
+
+		abciReq, ok := req.(*abci.RequestPrepareProposal)
+		if !ok {
+			return nil, fmt.Errorf("invalid request type: %T", req)
+		}
+
+		var maxBlockGas uint64
+
+		res, err := app.QueryWithState(ctx, rm, &consensusv1.QueryParamsRequest{})
```

Comment:
why not normal query? its unknown what actor/storekey is needed here

Reply:
We are receiving a reader, so I guessed we'd wanted to do this. Could there be a consensus engine that uses a height other than H-1 to create a proposal?

Reply:
the alternative would be normal query. Since consensus always and only deals with the latest state i would say its safe to assume it will always be the latest state

Reply:
Changed it to a normal Query with version 0. I also removed the store.Reader in the arguments of the func, as it's not needed then